[MRG+2] use manylinux dev wheels for numpy / scipy

matthew-brett opened the pull request:
Use daily manylinux wheels for numpy and scipy, instead of
soon-to-be-discontinued per-commit Precise wheels.

matthew-brett committed:
MAINT: use manylinux dev wheels for numpy / scipy

Use daily manylinux wheels for numpy and scipy, instead of
soon-to-be-discontinued per-commit Precise wheels.

jnothman commented:
LGTM, thanks

GaelVaroquaux commented:
Why did you remove the atlas libraries? It seems to me that an optimized blas would be useful. Do we have others on the system?

matthew-brett commented:
I removed the blas libraries thinking that you didn't need them anymore, now you are not using the numpy wheels that depend on them.  Is that not correct?

GaelVaroquaux commented:
I am pretty sure that we use a blas when it's available, for instance in sklearn/utils/weight_vector.pyx. We carry around our own blas headers (in sklearn/src/cblas) but I would rather rely on a proper blas implementation for building wheels.

What do you think? It seems to me that it would be better to have a good blas on the build system. Ideally openblas, though.

matthew-brett commented:
I may have misunderstood, but as far as I know the wheel-building repo is https://github.com/MacPython/scikit-learn-wheels - and you are not building wheels for redistribution here.

If you sometimes build with an external blas, and sometimes without, I guess the best option would be to have a build matrix entry for each option?

GaelVaroquaux commented:
> but as far as I know the wheel-building repo is https://github.com/MacPython/scikit-learn-wheels - and you are not building wheels for redistribution here.

Good point. I got confused.

> If you sometimes build with an external blas, and sometimes without, I guess the best option would be to have a build matrix entry for each option?

I'd rather avoid added one option to our build matrix, it's already big
enough, and consumming a lot of resources. The question is whether the
default should be with atlas or without.

If I compare the time of travis on your PR:
https://travis-ci.org/scikit-learn/scikit-learn/builds/207824847
to another recent build:
https://travis-ci.org/scikit-learn/scikit-learn
It seems to me that when there is atlas installed, the tests are
significantly faster (it could be something else, though). Hence, I am
more inclined to go for the atlas solution.

matthew-brett committed:
BF: add back ATLAS install for ubuntu build entry

scikit-learn can link against BLAS libraries still at the same location
as they were duing numpy build.

matthew-brett commented:
Hmm - looking at `sklearn/utils/setup.py`, I think you're only going to find the BLAS implementation when the library is in the same place as it was during the numpy build.  In practice, I think that's only going to apply to the Ubuntu build matrix entry, now, so I've tried to adapt `.travis.yml` to address that.

matthew-brett commented:
I reverted the change - so ATLAS is always installed now, as it was before this PR.

lesteve commented:
LGTM

raghavrv commented:
Thanks @matthew-brett :tada:
